Dockerfile.test

```diff
@@ -0,0 +1,25 @@
+FROM python:3.5
+
+WORKDIR /opt/fairing
+
+# Install dependencies
+COPY setup.py requirements.txt /opt/fairing/
```

Comment:
is this step redundant? line 23 would install deps too.

Reply:
No, this copies over the requirement files first to download the dependencies for more optimal caching. 

If done together with line 23, then every time the source changes, we'd redownload all the dependencies

Reply:
Yes, this helps us cache the dependencies in a different layer. The base image is not stored by default though, we'd need to add a separate step to pull the last image and use that as a cache - I'll go ahead and add that.

https://cloud.google.com/cloud-build/docs/speeding-up-builds

Reply:
Updated to cache test image across builds